src/main/java/uk/gov/hmcts/reform/civil/service/InitiateGeneralApplicationService.java

```diff
@@ -285,4 +327,61 @@ private List<String> getRespondentCaseRoles(CaseData caseData) {
         }
         return respondentCaseRoles;
     }
+
+    private Pair<CaseLocation, Boolean> getWorkAllocationLocation(CaseData caseData, String authToken) {
+        if (hasSDOBeenMade(caseData.getCcdState())) {
+            if (SPEC_CLAIM.equals(caseData.getSuperClaimType())) {
+                if (INDIVIDUAL.equals(caseData.getApplicant1().getType())
+                        || SOLE_TRADER.equals(caseData.getApplicant1().getType())) {
+                    return Pair.of(getClaimant1PreferredLocation(caseData), false);
+                } else {
+                    return Pair.of(getDefendant1PreferredLocation(caseData), false);
+                }
+            } else {
+                return Pair.of(getClaimant1PreferredLocation(caseData), false);
+            }
+        } else {
+            LocationRefData ccmccLocation = locationRefDataService.getCcmccLocation(authToken);
+            CaseLocation courtLocation = CaseLocation.builder()
+                    .region(ccmccLocation.getRegionId())
+                    .baseLocation(ccmccLocation.getEpimmsId())
+                    .build();
+            return Pair.of(courtLocation, true);
+        }
+    }
+
+    private boolean hasSDOBeenMade(CaseState state) {
+        return !statesBeforeSDO.contains(state);
+    }
+
+    private CaseLocation getClaimant1PreferredLocation(CaseData caseData) {
+        if (caseData.getApplicant1DQ() == null
+                || caseData.getApplicant1DQ().getApplicant1DQRequestedCourt() == null
+                || caseData.getApplicant1DQ().getApplicant1DQRequestedCourt().getResponseCourtCode() == null) {
+            return CaseLocation.builder().build();
+        }
+        return CaseLocation.builder()
+                //.region(caseData.getApplicant1DQ().getApplicant1DQRequestedCourt().getHearingPreferredRegionId())
+                .region(caseData.getApplicant1DQ().getApplicant1DQRequestedCourt().getResponseCourtCode())
+                //.baseLocation(caseData.getApplicant1DQ().getApplicant1DQRequestedCourt()
+                //.getHearingPreferredCourtVenueId())
```

Comment:
this requires modification

Reply:
based on the dq courtfinder changes  https://github.com/hmcts/civil-service/pull/1374/files

Reply:
Done